libs/core/src/lib/select/select.component.ts

```diff
@@ -142,98 +170,154 @@ export class SelectComponent implements OnChanges, AfterContentInit, OnInit, OnD
     @Output()
     readonly valueChange: EventEmitter<any> = new EventEmitter<any>();
 
+    /** @hidden */
+    @ContentChildren(OptionComponent, {descendants: true})
+    options: QueryList<OptionComponent>;
+
+    /** @hidden */
+    @ViewChild('selectControl')
+    controlElementRef: ElementRef;
+
+    /** Reference to root element for the mobile mode dialog */
+    @ViewChild('dialogContainer')
+    dialogContainerElementRef: ElementRef;
+
+    /** Reference to element containing list of options */
+    @ViewChild('selectOptionsListTemplate')
+    selectOptionsListTemplate: TemplateRef<any>;
+
     /** @hidden */
     calculatedMaxHeight: number;
 
     /** Current selected option component reference. */
-    private _selected: OptionComponent;
-
-    /** Subject triggered when the component is destroyed. */
-    private readonly _destroy$: Subject<void> = new Subject<void>();
-
-    /** Observable triggered when an option has its selectedChange event fire. */
-    private readonly _optionsStatusChanges: Observable<OptionComponent> = defer(() => {
-        const options = this.options;
-        if (options) {
-            return options.changes.pipe(
-                startWith(options),
-                switchMap(() => merge(...options.map((option) => option.selectedChange)))
-            );
-        }
-    }) as Observable<OptionComponent>;
+    selected: OptionComponent;
+
+    /** @hidden */
+    private _focusTrap: FocusTrap;
+
+    /** @hidden */
+    private _subscriptions: Subscription = new Subscription();
 
     /** @hidden */
     onChange: Function = () => {};
 
     /** @hidden */
     onTouched: Function = () => {};
 
-    constructor(private _changeDetectorRef: ChangeDetectorRef, @Optional() private _rtlService: RtlService) {}
+    /** @hidden */
+    @HostListener('keydown', ['$event'])
+    keydownHandler(event: KeyboardEvent): void {
+        if (isKey(event, 'ArrowUp')) {
+            if (this.isInteractive) {
+                this._focusOption('previous');
+            }
+            event.preventDefault();
+
+        } else if (isKey(event, 'ArrowDown')) {
+            if (this.isInteractive) {
+                this._focusOption('next');
+            }
+            event.preventDefault();
+
+        } else if (isKey(event, 'Escape')) {
+            if (this.isInteractive) {
+                this.close();
+            }
+            event.preventDefault();
+
+        } else if (isKey(event, ' ') || isKey(event, 'Enter')) {
+            if (this.isInteractive) {
+                this.toggle();
+            }
+            event.preventDefault();
+        }
+    }
+
+    /** @hidden */
+    @HostListener('window:resize')
+    resizeScrollHandler(): void {
+        this.calculatedMaxHeight = window.innerHeight * 0.45;
+    }
```

Comment:
Could you add this method also to `ngAfterViewInit`? calcualtedMaxHeight is only, when someone resize window